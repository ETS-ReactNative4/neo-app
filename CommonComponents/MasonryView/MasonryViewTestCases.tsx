import React, { useState, useEffect } from "react";
import { ITestCase } from "../../TypeInterfaces/TestCases/ITestCases";
import MasonryView from "./MasonryView";
import { IMasonryViewData } from "../../Screens/TravelProfileCityScreen/TravelProfileCity";
import { StyleSheet } from "react-native";
import {
  responsiveWidth
  // @ts-ignore
} from "react-native-responsive-dimensions";
import SelectablePortraitImage from "../SelectablePortraitImage/SelectablePortraitImage";

const data: IMasonryViewData[] = [
  {
    image:
      "https://pyt-images.imgix.net/images/city/paris.jpg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/city/london.jpg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/city/zagreb.jpg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/misc/germany.jpeg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/city/paris.jpg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/city/london.jpg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/city/zagreb.jpg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/misc/germany.jpeg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/city/paris.jpg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/city/london.jpg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/city/zagreb.jpg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/misc/germany.jpeg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/city/paris.jpg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/city/london.jpg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/city/zagreb.jpg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/misc/germany.jpeg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/city/paris.jpg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/city/london.jpg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/city/zagreb.jpg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/misc/germany.jpeg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/city/paris.jpg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/city/london.jpg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/city/zagreb.jpg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  },
  {
    image:
      "https://pyt-images.imgix.net/images/misc/germany.jpeg?w=146&h=200&fit=crop&dpr=2&auto=format,compress,enhance&q=20"
  }
];

const styles = StyleSheet.create({
  scrollColumn: {
    width: responsiveWidth(50) - 24
  },

  oddColumnStyle: {
    paddingLeft: 8
  },

  evenColumnStyle: {
    paddingRight: 8
  },
  selectablePortraitImageStyle: {
    marginBottom: 16
  }
});

interface MasonryViewWrapperProps {
  cityData: IMasonryViewData[];
}

interface ISuggestedCity {
  index: number;
  imageUrl: string;
  isSelected: boolean;
}

const MasonryViewWrapper = ({ cityData }: MasonryViewWrapperProps) => {
  const [suggestedCities, setSuggestedCities] = useState<ISuggestedCity[]>([]);

  useEffect(() => {
    setSuggestedCities(
      cityData.map((city, cityIndex) => {
        return {
          index: cityIndex,
          imageUrl: city.image,
          isSelected: false
        };
      })
    );
  }, [cityData]);

  const selectSuggestedCity = (cityIndex: number) => {
    const citiesList = [...suggestedCities];
    citiesList[cityIndex].isSelected = !citiesList[cityIndex].isSelected;
    setSuggestedCities(citiesList);
  };

  return (
    <MasonryView
      columns={2}
      columnStyle={styles.scrollColumn}
      oddColumnStyle={styles.oddColumnStyle}
      evenColumnStyle={styles.evenColumnStyle}
    >
      {suggestedCities.map((suggestedCity, suggestedCityIndex) => {
        const onSelect = () => {
          selectSuggestedCity(suggestedCity.index);
        };

        return (
          <SelectablePortraitImage
            key={suggestedCityIndex}
            onPress={onSelect}
            isSelected={suggestedCity.isSelected}
            imageSource={suggestedCity.imageUrl}
            containerStyle={styles.selectablePortraitImageStyle}
          />
        );
      })}
    </MasonryView>
  );
};

const SelectablePortraitImageTestCases: ITestCase[] = [
  {
    title: "Masonry View",
    Component: <MasonryViewWrapper cityData={data} />
  }
];

export default SelectablePortraitImageTestCases;
