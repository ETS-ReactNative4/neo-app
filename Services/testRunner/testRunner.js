import renderer from "react-test-renderer";

const testRunner = testCase => {
  test(testCase.title, () => {
    const tree = renderer.create(testCase.Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export default testRunner;
