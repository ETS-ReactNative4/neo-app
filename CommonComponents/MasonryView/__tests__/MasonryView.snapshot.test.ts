import MasonryViewTestCases from "../MasonryViewTestCases";
import testRunner from "../../../Services/testRunner/testRunner";

MasonryViewTestCases.map(testRunner);
