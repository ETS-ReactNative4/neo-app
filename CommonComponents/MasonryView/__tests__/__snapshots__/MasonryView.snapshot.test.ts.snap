// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Masonry View 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <ScrollView
    contentContainerStyle={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
    showsVerticalScrollIndicator={false}
  >
    <View />
  </ScrollView>
</View>
`;
